kie-server-parent/kie-server-wars/kie-server/src/main/webc-resources/README.md

```diff
@@ -114,26 +126,28 @@ This instruction describes all steps to install KIE Server on Tomcat 9 standalon
 
  8. Configure XA Recovery
 
-    Create xa recovery file next to the context.xml with data base configuration with following content:
+    Create an xa recovery file next to the context.xml called for example xa-recovery-properties.xml with data base configuration with following content:
 
+```
     <?xml version="1.0" encoding="UTF-8"?>
     <!DOCTYPE properties SYSTEM "http://java.sun.com/dtd/properties.dtd">
     <properties>
         <entry key="DB_1_DatabaseUser">sa</entry>
         <entry key="DB_1_DatabasePassword">sa</entry>
         <entry key="DB_1_DatabaseDynamicClass"></entry>
-        <entry key="DB_1_DatabaseURL">java:comp/env/h2DataSource</entry>
+        <entry key="DB_1_DatabaseURL">java:xads</entry>
     </properties>
+```
 
+    This causes the recovery process to utilise the underlying data source defined in the GlobalNamingResources section of the server.xml file.
     Append to CATALINA_OPTS in setenv.sh|bat file following:
     
     - setenv.sh:
-    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery\;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml\ \;1
+    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery\;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml
     
     - setenv.bat:
-    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml" ";1
-
+    -Dcom.arjuna.ats.jta.recovery.XAResourceRecovery1=com.arjuna.ats.internal.jdbc.recovery.BasicXARecovery;abs://$CATALINA_HOME/conf/xa-recovery-properties.xml"
 
     BasicXARecovery supports following parameters:
      - path to the properties file
-     - the number of connections defined in the properties file
+     - the number of connections defined in the properties file; this doesn't work until the fix for https://issues.jboss.org/browse/JBTM-3170 is incorporated into the release
```

Comment:
Currently narayana version is 5.9, where this bug was already fixed, so this clarification is no longer needed:
```
mvn dependency:tree | grep narayana-jta
[INFO] +- org.jboss.narayana.jta:narayana-jta:jar:5.9.0.Final:test
```

Reply:
ok, does it mean this setup was also verified by you @gmunozfe practically by tests or manually?
Because we talked with @rsynek that this probably still needs real verification

Reply:
@mareknovotny, I've tested with current libraries and followed the discussion from in the jira above.

As @SimonDallaway stated, there's an exception at startup due to JNDI _NameNotFoundException_, that doesn't happen when moving the configuration to **server.xml** under _GlobalNamingResources_. 

`WARN [Periodic Recovery] sun.reflect.NativeMethodAccessorImpl.invoke0 ARJUNA016009: Caught:
	java.sql.SQLException: javax.naming.NameNotFoundException: Name [comp/env/xads] is not bound in this Context. Unable to find [comp].`

That made me think this change was necessary, however after a deeper analysis this initial exception doesn't avoid _[Periodic Recovery]_ thread to be executed and worked out:

` [Periodic Recovery] com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule.xaRecoveryFirstPass xarecovery of com.mysql.jdbc.jdbc2.optional.JDBC4MysqlXAConnection@5a826a4b
 DEBUG [Periodic Recovery] com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule.xaRecoveryFirstPass Found 2 xids in doubt`

So, except moving the configuration from context.xml to web.xml (we can keep it as it is, and that part can be added as an alternative configuration), I agree with the rest of PR, because it adds some clarifications useful to the community.

I've tested it with mysql
`type="com.mysql.jdbc.jdbc2.optional.MysqlXADataSource"
`

and also extra configuration was needed in the setenv.sh script (specific for mysql):
`
-Dorg.kie.server.
persistence.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
`

Summarizing, @SimonDalaway, if you agree, perhaps you could rephrase it to keep both configurations, stating that setting the configuration at server.xml allows multiple applications running in the same Tomcat instance to share the database connection pool.

Reply:
Hi Gonzalo,

Tweaked it. Hopefully all good now.

Simon

Reply:
@gmunozfe So to sum up, this exception was just an initial warning and after that it works as well?

Reply:
Right @MarianMacik,
Exception is at first scan at [resourceInitiatedRecovery](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L194), but later, there's an additional scan adding those from [resourceInitiatedRecoveryForRecoveryHelpers](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L196) which retrieves correctly the resources (XAConnection) and therefore, works at [xaRecoveryFirstPass](https://github.com/jbosstm/narayana/blob/5.9.0.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L201)

Reply:
thanks guys for clarification and review